src/ui/analysis.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import moment from 'moment';
+import DataStoreContext from './datastore';
+
+/**
+ * A singleton class for calculating analysis of a run.
```

Comment:
Importing this class seems to create a new instance every time. Is it really a singleton?

Reply:
I add a `console.log` before the `export default`, and it only gets called once.  That matches my understanding of how importing / requiring works -- a module's definition is only executed once and the result cached by the require machinery.  It's similar to `export default () => {..}` -- every place you `import` or `require` that module will get the identical function object.

Reply:
You are right here. My mistake.